src/app/intel.cpp

```diff
@@ -0,0 +1,224 @@
+#include <nlohmann/json.hpp>
```

Comment:
Please keep the Intel License if you copied any line of code from the original code:

```
/*
 * Copyright © 2007-2021 Intel Corporation
 * Copyright © 2023  Kurt Kartaltepe
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and associated documentation files (the "Software"),
 * to deal in the Software without restriction, including without limitation
 * the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the
 * Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice (including the next
 * paragraph) shall be included in all copies or substantial portions of the
 * Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 * DEALINGS IN THE SOFTWARE.
 */
```

Reply:
The code submitted was authored entirely by me and is submitted reflecting that. Though their code was referenced as a working implementation of reading perf counters and precisely which perf counters they use.

If you still feel the implementation bears too much of a resemblance to some portions of their code I'm happy to attribute the copyright to intel.

Reply:
Great to hear! The MIT license is not the end of the world anyway, but clean code is I guess better anyway.

What license would you want to use for your implementation?

Reply:
Feel free to use all code in this PR under the same license as MangoHud, which appears to be MIT.

Reply:
FWIW note that since a few kernels ago it is possible to query individual DRM file descriptor GPU usage via /proc/<pid|self>/fdinfo/<fd>. That would not require any extra capabilities for own file descriptors (assuming with LD_PRELOAD you are in this situation). Downside is it only gives you own GPU usage, no global stats frequency, power etc. Nevertheless might be even interesting to see how much a game is using separately from maybe game + compositor. 

P.S. AMD supports the same interface.